cdisc_rules_engine/check_operators/sql_operators.py

```diff
@@ -1126,21 +1126,32 @@ def compare_target_with_comparator_next_row(self, df: DatasetInterface, target:
     @type_operator(FIELD_DATAFRAME)
     def present_on_multiple_rows_within(self, other_value: dict):
         """
-        The operator ensures that the target is present on multiple rows
-        within a group_by column. The dataframe is grouped by a certain column
-        and the check is applied to each group.
+        Checks if a target value is present on multiple rows within a group.
         """
-        target = self.replace_prefix(other_value.get("target"))
-        min_count: int = other_value.get("comparator") or 1
-        group_by_column = self.replace_prefix(other_value.get("within"))
-        grouped = self.validation_df.groupby([group_by_column, target])
-        meta = (target, bool)
-        results = grouped.apply(lambda x: self.validate_series_length(x, target, min_count), meta=meta)
-        uuid = str(uuid4())
-        return self.validation_df.merge(results.rename(uuid).reset_index(), on=[group_by_column, target])[uuid]
+        target_column = self.replace_prefix(other_value.get("target")).lower()
+        min_count = other_value.get("comparator") or 1
+        within_column = self.replace_prefix(other_value.get("within")).lower()
+
+        op_name = f"{target_column}_{within_column}_{min_count}_present_on_multiple_rows"
+
+        def generate_update_query(db_table: str, db_column: str) -> str:
+            # Construct the UPDATE query. This uses a window function to count
+            # occurrences within each group and then updates each row accordingly.
+            return f"""
+                UPDATE {db_table} AS t
```

Comment:
Let's leave it like this for now, but we might have other check operators which do the same sort of transformation and we can refactor some of this later